### ISSUE TITLE
Dying mobs teleport away

### ISSUE THREAD

**Issue by Knight-Pachirisu** [Progressive335/Progressive335#361]
**Description of the bug:**

This issue has occurred several times before on several characters and on different places/mobs, though I noticed it more on Humanoids/mobs that can run away.
Sometimes, either in the open world or in dungeons, when the mobs are close to dying, instead of running/walking for the duration they should, they instead teleport away, from what I could see, to the starting position they were in when pulled.
After a second or so, they are teleported back to where they were before the teleport happened, but if there are any mobs close to where the first teleport took them to, these will aggro. This is particularly annoying when in dungeons that have patrols/not a lot of room to move around.

Also, worth noting, if they die, due to Damage over Time, for instance, they aren't teleported back.

**Steps to reproduce the problem:**

There wasn't any particular way to replicate this, that I was able to tell, it seems random, but it's always on dying mobs and it seems like those that run when close to death are far more likely to do so, if not the only ones that has this happening.

**How it should work:**

Have them only walk away.

**Any other documentation that could be helpful:**

I'll try to get a screenshot next time it happens.  
There is already another bug, similar to this in here, maybe the two are connected?

**Comment by LezChap** [comment]
So you aren't trying to stun them when they're close to dying?  They just teleport and die?

**Comment by Zooke** [comment]
Related issue, https://github.com/Progressive335/Progressive335/issues/346
Whats happening is that the server decides the final flee position of the mob, but the part where the mob runs there is not always working correctly given various circumstances.

**Comment by Knight-Pachirisu** [comment]
LezChap
Not always, no, when pulling lots of mobs and using AoE that's when I noticed it the most. And in this case it's rather complicated to stun them all, best to just keep the AoE going in hopes they'll die before they leave the area.

Zooke
Thanks for the info, the issue is that they do it right away.  In short, in the time that would normally take them to go from where they are to where they would have been after the run/walk, is much greater than the teleport.
Not only that, but they then teleport back to the same spot before the first teleport. So they jump back and forth once, if a DoT doesn't kill them in the small window of the first teleport.

I did see the vids and it's pretty much like what happened to you on the second one at 0:45
